### PR TITLE
Add linting capability.

### PR DESCRIPTION
**Pull request by jamiebrynes7**
This PR introduces a C# linter using `dotnet-format`. It provides:

- A docker image with the formatter installed.
- A generic lint script `script/lint.sh` for linting a given directory.
- Linting setup for this repository.

### CODE REVIEW

**Review by zeroZshadow — APPROVED** [review]

File: ci/lint.sh [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+
+set -e -u -o pipefail
+
+if [[ -n "${DEBUG-}" ]]; then
+  set -x
+fi
+
+cd "$(dirname "$0")/.."
+
+./scripts/lint.sh ./tools --check
```
Comment (paulbalaji) [inline ci/lint.sh]: why no newline at the end of here, but we keep it in `scripts/lint.sh`?

Comment (paulbalaji) [reply ci/lint.sh]: same question for the Dockerfile

**Review by paulbalaji — APPROVED** [review]

File: ci/lint.sh (hunk 1 above)
Comment (jamiebrynes7) [reply ci/lint.sh]: Forgot to add them 🤷‍♂

**Comment by paulbalaji** [comment]
fair enough lol

File: ci/lint.sh (hunk 1 above)
Comment (paulbalaji) [reply ci/lint.sh]: fair enough :P

File: ci/docker/lint/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+# Install the linting tool
+RUN dotnet tool install --global dotnet-format
+ENV PATH="${PATH}:/root/.dotnet/tools"
+
+# Setup the volume which we will be running from
+VOLUME /project
+WORKDIR /project
+
+# Entrypoint
+ENTRYPOINT [ "dotnet", "format" ]
```
Comment (jamiebrynes7) [inline ci/docker/lint/Dockerfile]:
```suggestion
ENTRYPOINT [ "dotnet", "format" ]

```

File: ci/lint.sh (hunk 1 above)
Comment (jamiebrynes7) [inline ci/lint.sh]:
```suggestion
./scripts/lint.sh ./tools --check

```

**Pull request merged**